bgpd/bgp_evpn.c

```diff
@@ -5391,8 +5391,10 @@ int bgp_filter_evpn_routes_upon_martian_nh_change(struct bgp *bgp)
 				if (!(pi->type == ZEBRA_ROUTE_BGP
 				      && pi->sub_type == BGP_ROUTE_NORMAL))
 					continue;
-
-				if (bgp_nexthop_self(bgp, pi->attr->nexthop)) {
+				/* TODO : Need to test these chnages */
+				if (bgp_nexthop_self(bgp, afi,
```

Comment:
only unicast is handled here?

Reply:
if you see at line no 5369 afi & safi are initialised AFI_L2VPN and SAFI_EVPN respectively.